firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -62,11 +62,19 @@
   private AppDistributionReleaseInternal cachedNewRelease;
 
   private Task<AppDistributionRelease> cachedCheckForNewReleaseTask;
-  private AlertDialog updateDialog;
+  private AlertDialog updateConfirmationDialog;
   private AlertDialog signInConfirmationDialog;
-  private boolean remakeUpdateDialog = false;
+  @Nullable private Activity dialogHostActivity = null;
+
+  @GuardedBy("updateIfNewReleaseTaskLock")
```

Comment:
I don't think either of these booleans need to be protected by this lock, or any other lock. They're almost always read or written from the main thread (in lifecycle callbacks), and in the case where they're being set to false in updateIfNewReleaseAvailable, it won't practically affect anything because there's no task in progress at that point.

Reply:
That is true I agree calling the updateIfNewRelease would not be an issue. I don't see a problem with onPause being called multiple times from different places but if those are all on the main thread there shouldn't be a deadlock issue so as long as we are sure of that we can change it.